### DISCUSSION TITLE
Where is the "admin" variant sorting saved?

### DISCUSSION THREAD

**Discussion by sergiobellini** [woocommerce/woocommerce#41698] [General]
Hi everyone, in my WC products I use variants created on non-terms attributes (values separated by |, example "M | S | XL | XXL | L"). Unfortunately, the use of terms creates other problems for me, as I import the data into WC from external tools.
On the product page the order of the variants follows this "M | S | XL | XXL | L" rather than following the order I drag-n-drop on the product edit page. I would like to know where that order is stored and whether via hook/filter/JS I can intervene to read and modify it on the product page.
Thanks everyone in advance.

**Comment by mattsherman** [comment]
Hi @sergiobellini,

It looks like you might be able to use the `woocommerce_dropdown_variation_attribute_options_args` filter to accomplish this. I did a quick test (after a much longer time digging into how to potentially accomplish this!) with the code below and it appeared to work as expected.

However, I'll be honest that I'm not sure this is the best approach. For one, it basically is hard-coding the attribute values. If that can be done, than I'd argue that the attribute values shouldn't be imported from an external source. Because, you'd have to update your filter every time the imported values changed (and, how would you know this?). Now, if you were doing a simple alphabetic sort, then you could do this without hard-coding the values.

And, I also admit that my knowledge on this is not extensive.

Others may have better solutions, perhaps even a block-based, no code approach. /cc  @woocommerce/woo-fse 

```php
<?php

add_filter(
	'woocommerce_dropdown_variation_attribute_options_args',
	function( $args ) {
		if ( 'Size' === $args['attribute'] ) {
			$args['options'] = array( 'S', 'M', 'L', 'XL', 'XXL' );
		}

		return $args;
	},
	10,
);
```

**Comment by sergiobellini** [comment]
Hi @mattsherman,
thanks for the reply, I will study your solution. In some cases I can definitely order the entries this way, in others I go alphabetically, that's a solution. However, I would have liked to understand where the order of the variants is stored, which can be done via drag-n-drop in the product edit...
